tic-tac-toe.js

```diff
@@ -1,7 +1,86 @@
 function TicTacToe() {
-
+  this.board = [[0, 0, 0],
+                [0, 0, 0],
+                [0, 0, 0]
+                ];
+  this.player_1 = new Player(1, "Ham", "🐷");
+  this.player_2 = new Player(2, "Abu", "🐵");
+  this.current_player = this.player_1;
+  this.winning_sequence = null;
+  this.turnCount = 0;
 }
 
-TicTacToe.prototype = {
-  
+function Player(number, name, marker) {
+  this.number = number;
+  this.name = name;
+  this.marker = marker;
 }
+
+TicTacToe.prototype.playTurn = function(row, column) {
+  this.board[row][column] = this.current_player.number;
+  this.checkWinningBoard();
+  this.turnCount++;
+  if (this.winning_sequence) {
+      $("td").empty();
+      $(".board_footer").empty();
+      $(".row_" + this.winning_sequence[0][0] + ".column_" + this.winning_sequence[0][1]).text(this.current_player.marker);
+      $(".row_" + this.winning_sequence[1][0] + ".column_" + this.winning_sequence[1][1]).text(this.current_player.marker);
+      $(".row_" + this.winning_sequence[2][0] + ".column_" + this.winning_sequence[2][1]).text(this.current_player.marker);
+      $( "table").after("<h1>" + this.current_player.name + " Wins!" + "</h1>");
+  } else if (this.turnCount === 9) {
+    $(".board_footer").empty();
+    $( "table").after("<h1>Game Over</h1>");
+  } else {
+    this.switch_player();
+  }
+};
+
+TicTacToe.prototype.switch_player = function() {
+  if (this.current_player === this.player_1) {
+    this.current_player = this.player_2;
+  } else {
+    this.current_player = this.player_1;
+  }
+};
+
+TicTacToe.prototype.checkWinningBoard = function() {
+  if ((this.board[0][0] === this.current_player.number) && (this.board[0][1] === this.current_player.number) && (this.board[0][2] === this.current_player.number)) {
```

Comment:
I wonder if you could put all of these row/col values in an array and iterate over it to slightly more dynamically create this winning sequence instead of using this many conditionals.